src/requestforattestation/RequestForAttestation.ts

```diff
@@ -61,6 +63,95 @@ function getHashRoot(leaves: Uint8Array[]): Uint8Array {
   return hash(result)
 }
 
+function compressNonceAndHash(nonceHash: any[]): any {
+  if (nonceHash.length === 0) {
+    return {
+      hash: nonceHash[0],
+    }
+  }
+  return {
+    nonce: nonceHash[0],
+    hash: nonceHash[1],
+  }
+}
+
+export function compressClaimHashTree(
+  rfa: IRequestForAttestation
+): IRequestForAttestation['claimHashTree'] {
+  const result = {}
+  const claimTree = rfa.claimHashTree
+
+  Object.keys(claimTree).forEach(entryKey => {
+    result[entryKey] = Object.values(claimTree[entryKey])
+  })
+  return result
+}
+
+export function decompressClaimHashTree(
+  rfa: IRequestForAttestation[]
+): IRequestForAttestation['claimHashTree'] {
+  const result = {}
+
+  Object.keys(rfa).forEach(entryKey => {
+    result[entryKey] = compressNonceAndHash(Object.values(rfa[entryKey]))
+  })
+  return result
+}
+
+export function compressClaimContents(rfa: IRequestForAttestation): any {
+  // should go into the claim module.
```

Comment:
Question:
Is this comment intended here? Or maybe you want to make this change?

Reply:
I was leaving this for a place holder to ask.

Reply:
Do you think this should go inside the claim class?

Reply:
I see you've updated this 👍